source/server/options_impl.cc

```diff
@@ -48,6 +48,9 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const HotRestartVersionCb& hot_r
                                         std::thread::hardware_concurrency(), "uint32_t", cmd);
   TCLAP::ValueArg<std::string> config_path("c", "config-path", "Path to configuration file", false,
                                            "", "string", cmd);
+  TCLAP::ValueArg<std::string> config_json(
+      "", "config-json", "Json configuration, only for v2, overrides --config-path", false, "",
```

Comment:
Thinking about this behavior, another option would be to merge on top of whatever we read from `--config-path` via proto `MergeFrom`. This would provide a very flexible and powerful way to do CLI overrides, which would be cheap to implement. WDYT?

Reply:
TBH I would rather keep things simple. If you feel strongly about this I'm happy to implement though. This could also be done later if someone needed this without changing the flags.

Reply:
Done, but it is a bit more than that since I want to allow to make the --config-path optional.